.github/workflows/sonarcloud.yml

```diff
@@ -0,0 +1,50 @@
+name: SonarQube Analysis
+on:
+  schedule:
+    - cron: '0 1 * * 0'
+  workflow_dispatch: {}
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  build:
+    name: Build and analyze
+    runs-on: windows-latest
+    steps:
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'zulu'
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Cache SonarQube Cloud packages
```

Comment:
Is this cache needed for SonarQube? I'd prefer less github actions, if the caching is not needed (if gains are minimal)

Reply:
The workflow follows the recommended steps from SonarQube Cloud's CI-based analysis.